src/components/Pages/RequestPage/RequestPage.tsx

```diff
@@ -16,6 +16,49 @@ export const RequestPage = () => {
   const [resultSent, setResultSent] = useState<boolean>(false)
   const [denied, setDenied] = useState<boolean>(false)
 
+  const getProvider = async () => {
```

Comment:
Would you mind moving this outside of the component or using the `useCallback` react method?